app/src/main/java/io/homeassistant/companion/android/widgets/template/TemplateWidgetConfigureActivity.kt

```diff
@@ -216,10 +216,10 @@ class TemplateWidgetConfigureActivity : BaseWidgetConfigureActivity() {
             var enabled: Boolean
             withContext(Dispatchers.IO) {
                 try {
-                    templateText = integrationUseCase.renderTemplate(template, mapOf())
+                    templateText = integrationUseCase.renderTemplate(template, mapOf()) ?: getString(commonR.string.template_error)
                     enabled = true
                 } catch (e: Exception) {
-                    templateText = "Error in template"
+                    templateText = getString(commonR.string.error_connection_failed)
```

Comment:
I'm not sure this is a good error message. When starting to type brackets on a phone, after typing "{{" the preview shows "Unable to connect to Home Assistant." while the problem isn't related to the network but to the template text.

Server response for "{{" is: `{"template": {"error": "TemplateSyntaxError: unexpected 'end of template'"}}`

Reply:
Oh, didn't test it properly then. And annoying that it sometimes gives an error message and sometimes null... I'll update it somewhere this weekend